yarn-project/simulator/src/avm/opcodes/external_calls.ts

```diff
@@ -69,21 +71,25 @@ abstract class ExternalCall extends Instruction {
     const totalGas = sumGas(this.gasCost(memoryOperations), allocatedGas);
     context.machineState.consumeGas(totalGas);
 
-    // TRANSITIONAL: This should be removed once the AVM is fully operational and the public executor is gone.
+    // TRANSITIONAL: This should be removed once the kernel handles and entire enqueued call per circuit
     const nestedContext = context.createNestedContractCallContext(
       callAddress.toFr(),
       calldata,
       allocatedGas,
       this.type,
       FunctionSelector.fromField(functionSelector),
     );
-    const pxContext = createPublicExecutionContext(nestedContext, calldata);
-    const pxResults = await executePublicFunction(pxContext, /*nested=*/ true);
+
+    const startPxContext = createPublicExecutionContext(nestedContext, calldata);
```

Comment:
Question about naming: does the `startPxContext` stay constant or does it indirectly get modified during the simulator execution (e.g., if things are copied [by reference] from the nestedContext).

Reply:
You are right that it's copied by reference.... We only use `startPxContext.{execution,availableGas,transactionFee}`, so it shouldn't be a problem. But maybe the conversion function should just accept those directly :thinking: